<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Downloading and Parsing a Genome</title>

<script src="site_libs/header-attrs-2.8/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Nathan's Projects</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Visualizing Arlington Bikeometers
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="part_1_the_plan.html">Part 1: The Plan</a>
    </li>
    <li>
      <a href="part_2_query_the_api.html">Part 2: Query the API</a>
    </li>
    <li>
      <a href="part_3_setup_your_database.html">Part 3: Setup and Connect Your Database</a>
    </li>
    <li>
      <a href="part_4_save_the_data_in_a_mysql_database.html">Part 4: Save the Data in Your Local MySQL Database</a>
    </li>
    <li>
      <a href="part_5_picking_bikeometers_to_graph.html">Part 5: Picking Which Bikeometers to Graph</a>
    </li>
    <li>
      <a href="part_6_creating_initial_visualizations.html">Part 6: Creating the Visualizations</a>
    </li>
    <li>
      <a href="part_7_investigating_bikometer_gps_locations.html">Part 7: Investigating Bikeometer GPS Locations</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Bioinformatics Notes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Running-BLAST-over-the-internet.html">Running BLAST Over the Internet</a>
    </li>
    <li>
      <a href="communicating_with-the_outside.html">Communicating with the Outside</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Notes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="note_fixing_pyrtools_error.html">Fixing rpytools error</a>
    </li>
    <li>
      <a href="note_knit_rmarkdown_db.html">Publish Rmarkdown Documents With Database Connections Without Exposing Credentials</a>
    </li>
    <li>
      <a href="note_general_r_notes.html">General R Notes</a>
    </li>
  </ul>
</li>
<li>
  <a href="photography.html">Photography</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/photonathan">Github</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Downloading and Parsing a Genome</h1>

</div>


<p>Read the genome into memory.</p>
<pre class="python"><code>def readGenome(filename):
  genome = &#39;&#39;
  with open(filename, &#39;r&#39;) as f:
    for line in f:
      if line[0] != &#39;&gt;&#39;:
        # rstrip method removes white spaces and new line at end of line
        genome += line.rstrip()
  return genome

genome = readGenome(&#39;data/lambda_virus.fa&#39;)
genome[:100]</code></pre>
<pre><code>## &#39;GGGCGGCGACCTCGCGGGTTTTCGCTATTTATGAAAATTTTCCGGTTTAAGGCGTTTCCGTTCTTCTTCGTCATAACTTAATGTTTTTATTTAAAATACC&#39;</code></pre>
<p>Count the frequency of each base using a dictionary.</p>
<pre class="python"><code>counts = {&#39;A&#39;: 0, &#39;C&#39;: 0, &#39;G&#39;: 0, &#39;T&#39;: 0}
for base in genome:
  counts[base] += 1
print(counts)</code></pre>
<pre><code>## {&#39;A&#39;: 12334, &#39;C&#39;: 11362, &#39;G&#39;: 12820, &#39;T&#39;: 11986}</code></pre>
<p>Use the <em>collections</em> module to count the bases.</p>
<pre class="python"><code>import collections
collections.Counter(genome)</code></pre>
<pre><code>## Counter({&#39;G&#39;: 12820, &#39;A&#39;: 12334, &#39;T&#39;: 11986, &#39;C&#39;: 11362})</code></pre>
<p>To read a FASTQ file and pull out the sequences and quality scores, use the below function:</p>
<pre class="python"><code>def readFastq(filename):
  sequences = []
  qualities = []
  with open(filename) as fh:
    while True:
      # Read this line but dont save it
      fh.readline()
      # Read this line and save
      seq = fh.readline().rstrip()
      fh.readline()
      qual = fh.readline().rstrip()
      # If you are at the end of the file, break
      if len(seq) == 0:
        break
      sequences.append(seq)
      qualities.append(qual)
  return sequences, qualities</code></pre>
<pre class="python"><code>seqs, quals = readFastq(&#39;data/ERR037900_1.first1000.fastq&#39;)</code></pre>
<pre class="python"><code>print(seqs[:5])</code></pre>
<pre><code>## [&#39;TAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCNAACCCTAACCCTAACCCTAACCCTAACCCTAAC&#39;, &#39;TAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCNAACCCTAACCCTAACCCTAACCCTNACCCTAAC&#39;, &#39;TAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCNAACCCTAACCCTAACCCTAACCCTAACCCTAAC&#39;, &#39;TAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCNAACCCTAACCCTAACCCTAACCCTAACCCTACC&#39;, &#39;AACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTNACCCTAACCCTAACCCTAACCCTAAACCTAACC&#39;]</code></pre>
<pre class="python"><code>print(&#39;&#39;)</code></pre>
<pre class="python"><code>print(quals[:5])</code></pre>
<pre><code>## [&#39;HHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHGFHHHFHFFHHHHHGHHFHEH@4#55554455HGFBF&lt;@C&gt;7EEF@FBEDDD&lt;=C&lt;E&#39;, &#39;HHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHCHHHHEHHBA#C&gt;@54455C/7=CGHEGEB;C############&#39;, &#39;HHHHHHHHHHHHHHHHHHHHHHHHHHHHHHGHHHHHHHHHHHHHDHHHDEHHHHFGIHEHEGGGF4#45655366GIGEHAGBG################&#39;, &#39;HHHHHHHHHHHHHHHHHHHHHHHHHIHHHHHHHHHHHHHHHHHHHHHHIHHHHHIHFHHHIHHHHD#ECA54655GGIBH?BD@+BCBF?5A=::&gt;8?##&#39;, &#39;HHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHIHIHEHHIGHIFFHIIGF6#555:2=7=CB;?3CAACBAC2B###########&#39;]</code></pre>
<p>In a FASTQ file, you can convert a character to an integer by using the following code:</p>
<pre class="python"><code>def QtoPhred33(Q):
  &#39;&#39;&#39; Turn Q into Phred+33 ASCII-encoded quality &#39;&#39;&#39;
  return chr(Q + 33)

def phred33ToQ(qual):
  &#39;&#39;&#39; Turn Phred+33 ASCII-encoded quality into Q &#39;&#39;&#39;
  # ord() takes a string and converts it to and integer equivalent
  return ord(qual)-33</code></pre>
<pre class="python"><code>phred33ToQ(&#39;#&#39;)</code></pre>
<pre><code>## 2</code></pre>
<p>‘2’ is a low quality score and means that the program has low confidence that this is the correct base.</p>
<p>Next, we will create a histogram of the quality scores.</p>
<pre class="python"><code>def createHist(qualities):
  hist = [0] * 50
  for qual in qualities:
    for phred in qual:
      q = phred33ToQ(phred)
      hist[q] += 1
  return hist</code></pre>
<pre class="python"><code>h = createHist(quals)</code></pre>
<pre class="python"><code>h</code></pre>
<pre><code>## [0, 0, 17723, 0, 2, 11, 11, 28, 23, 55, 100, 111, 86, 174, 185, 272, 317, 259, 390, 1523, 2782, 762, 286, 413, 403, 538, 351, 694, 971, 777, 1024, 1449, 1341, 1312, 1916, 2233, 3025, 4043, 6640, 45696, 2074, 0, 0, 0, 0, 0, 0, 0, 0, 0]</code></pre>
<pre class="python"><code>type(h)</code></pre>
<pre><code>## &lt;class &#39;list&#39;&gt;</code></pre>
<pre class="r"><code>library(reticulate)
py$h</code></pre>
<pre><code>##  [1]     0     0 17723     0     2    11    11    28    23    55   100   111
## [13]    86   174   185   272   317   259   390  1523  2782   762   286   413
## [25]   403   538   351   694   971   777  1024  1449  1341  1312  1916  2233
## [37]  3025  4043  6640 45696  2074     0     0     0     0     0     0     0
## [49]     0     0</code></pre>
<pre class="r"><code>library(ggplot2)
library(dplyr)</code></pre>
<pre class="r"><code>df &lt;- data.frame(&#39;quality score&#39; = c(1:50), &#39;quantity&#39; = py$h)</code></pre>
<pre class="r"><code>df</code></pre>
<pre><code>##    quality.score quantity
## 1              1        0
## 2              2        0
## 3              3    17723
## 4              4        0
## 5              5        2
## 6              6       11
## 7              7       11
## 8              8       28
## 9              9       23
## 10            10       55
## 11            11      100
## 12            12      111
## 13            13       86
## 14            14      174
## 15            15      185
## 16            16      272
## 17            17      317
## 18            18      259
## 19            19      390
## 20            20     1523
## 21            21     2782
## 22            22      762
## 23            23      286
## 24            24      413
## 25            25      403
## 26            26      538
## 27            27      351
## 28            28      694
## 29            29      971
## 30            30      777
## 31            31     1024
## 32            32     1449
## 33            33     1341
## 34            34     1312
## 35            35     1916
## 36            36     2233
## 37            37     3025
## 38            38     4043
## 39            39     6640
## 40            40    45696
## 41            41     2074
## 42            42        0
## 43            43        0
## 44            44        0
## 45            45        0
## 46            46        0
## 47            47        0
## 48            48        0
## 49            49        0
## 50            50        0</code></pre>
<p>Don’t use geom_bar(), use geom_col(). The function geom_bar() makes the height of the bar proportinal to the number of cases in each group, or if the weight asthetic is supplied, the sum of the weights. If you want hights of the bars to represent values in the data, use geom_col().</p>
<pre class="r"><code>p &lt;- df %&gt;% ggplot(aes(x = quality.score, y = quantity)) + geom_col()
p</code></pre>
<p><img src="downloading_and_parsing_a_genome_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
